packages/cojson/src/PriorityBasedMessageQueue.ts

```diff
@@ -82,35 +85,79 @@ export class LinkedList<T> {
 
     this.length--;
 
+    this.meter?.pull();
     return value;
   }
 }
 
+class QueueMeter {
+  private pullCounter: Counter;
+  private pushCounter: Counter;
+
+  constructor(
+    prefix: string,
+    private attrs?: Record<string, string | number>,
+  ) {
+    this.pullCounter = metrics
+      .getMeter("cojosn")
+      .createCounter(`${prefix}.pulled`, {
+        description: "Number of messages pulled from the queue",
+        valueType: ValueType.INT,
+        unit: "1",
+      });
+    this.pushCounter = metrics
+      .getMeter("cojosn")
+      .createCounter(`${prefix}.pushed`, {
+        description: "Number of messages pushed to the queue",
+        valueType: ValueType.INT,
+        unit: "1",
+      });
+
+    this.pullCounter.add(0, this.attrs);
+    this.pushCounter.add(0, this.attrs);
```

Comment:
this makes sure that those metrics are generated (and emitted) as soon as the queue is created. this is to avoid edge cases where one series reset is delayed, which would cause spikes or dips when queried - and it also more correctly represents the actual state of the queue after a restart.

Reply:
Great explaination, thanks!

Could you add it into the code as a comment?

Reply:
yep. good point. (is also covered by a test btw)